sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFnReflector.java

```diff
@@ -552,8 +576,17 @@ public String subclass(Generic superClass) {
           .intercept(InvokerDelegation.create(
               startBundle, BeforeDelegation.INVOKE_PREPARE_FOR_PROCESSING, startBundleArgs))
           .method(ElementMatchers.named("invokeFinishBundle"))
-          .intercept(InvokerDelegation.create(
-              finishBundle, BeforeDelegation.NOOP, finishBundleArgs));
+          .intercept(InvokerDelegation.create(finishBundle,
+              BeforeDelegation.NOOP,
+              finishBundleArgs))
+          .method(ElementMatchers.named("invokeSetup"))
+          .intercept(InvokerDelegation.create(setup,
+              BeforeDelegation.NOOP,
+              Collections.<AdditionalParameter>emptyList()))
+          .method(ElementMatchers.named("invokeTeardown"))
```

Comment:
Should add at least one test that verifies invokeSetup/invokeTeardown properly invoke the expected method, etc.